### PR TITLE
Add possibility to choose vacuum and/or repack 

### PR DESCRIPTION
**Pull request by wejdross**
This MR adds necessary logic for https://github.com/vshn/appcat/pull/158 . Additionally it's shipped with e2e tests. 

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/component/vshn_services.jsonnet [hunk 1]
```diff
@@ -30,14 +30,29 @@ local maintenanceRole = kube.ClusterRole(maintenanceRoleName) {
...
 
+
 local maintenanceClusterRoleBinding = kube.ClusterRoleBinding('crossplane:appcat:job:helm:maintenance') + {
   roleRef_: maintenanceRole,
   subjects_: [ maintenanceServiceAccount ],
 };
 
+// this role is necessary for maintenance logic to get info if vacuum or repack is enabled in .spec.parameters.service
+// clusterrolebinding is created as object and grant permissions to maintenance service account in instance namespace
+// don't add LIST verb here, maintenance service account could in some specific cases be reused by customers i shared clusters
```
Comment (Kidswiss) [inline component/component/vshn_services.jsonnet:41]: We shouldn't really add any other verbs here. If we ever need more permissions, I would strongly suggest to move to the appcat-contol namespace, like we do with the helmpatcher.

The `get` verb should be fine, there's not much a customer can do with it. Although guessing names for claims might not be too hard, as they don't have any random suffixes.

What do you think @zugao

File: component/component/provider.jsonnet [hunk 2]
```diff
@@ -173,8 +173,8 @@ local runtimeConfigRef(name) = {
         },
         {
           apiGroups: [ 'rbac.authorization.k8s.io' ],
-          resources: [ 'clusterrolebindings', 'roles', 'rolebindings' ],
-          verbs: [ 'get', 'list', 'watch', 'update', 'patch', 'create', 'delete' ],
+          resources: [ 'clusterrolebindings', 'roles', 'rolebindings', 'clusterroles' ],
+          verbs: [ '*' ],
```
Comment (Kidswiss) [inline component/component/provider.jsonnet:177]: Please don't use `'*'` for verbs. We'll get flagged by automated security scanners with this.

File: tests/e2e/common/scripts/trigger-maintenance.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+set -e
+
+type="$1"
+name="$2"
+to_be_found="$3"
+
+ns=$(kubectl -n "$NAMESPACE" get "$type" "$name" -oyaml | yq -r '.status.instanceNamespace')
+
+kubectl -n "$ns" delete job --ignore-not-found=true --now=true vacuum databasesrepack
```
Comment (Kidswiss) [inline tests/e2e/common/scripts/trigger-maintenance.sh:11]: Did you test against the lab? I'm pretty sure you'll run into issues as the e2e user doesn't have the permissions for jobs. You need to add them in `rbac_e2e.jsonnet`.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Some minor things.

I'd also want to have gabriel on the same page about the implications of the clusterrole. I personally would have moved it to the `appcat-control` namespace, so that's tighter and harder to abuse.

**Comment by wejdross** [comment]
@Kidswiss testing in LAB was a little bit of a pain but with tiny script I was able to make that work. Please take a second look. I addressed all You remarks. E2E are passing in kind as well as LAB cluster.

File: tests/e2e/common/scripts/trigger-maintenance.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+# wait for cronjob to receive updated environment variables - it takes few reconciliations to update the envs
+if [ "$to_be_found" == "b" ]; then
+  while true; do
+    vacuum_enabled=$(kubectl get -n "$ns" cronjobs.batch --ignore-not-found maintenancejob -o jsonpath='{.spec.jobTemplate.spec.template.spec.containers[0].env[?(@.name=="VACUUM_ENABLED")].value}')
+    repack_enabled=$(kubectl get -n "$ns" cronjobs.batch --ignore-not-found maintenancejob -o jsonpath='{.spec.jobTemplate.spec.template.spec.containers[0].env[?(@.name=="REPACK_ENABLED")].value}')
+    if [ "$vacuum_enabled" == "true" ] && [ "$repack_enabled" == "true" ]; then
+      break
+    fi
+    echo "Waiting for envs: True True"
```
Comment (Kidswiss) [inline tests/e2e/common/scripts/trigger-maintenance.sh:21]:
```suggestion
    echo "Waiting for envs: vacuum: True repack: True"
```

File: tests/e2e/common/scripts/trigger-maintenance.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    sleep 15
+  done
+elif [ "$to_be_found" == "v" ]; then
+  while true; do
+    vacuum_enabled=$(kubectl get -n "$ns" cronjobs.batch --ignore-not-found maintenancejob -o jsonpath='{.spec.jobTemplate.spec.template.spec.containers[0].env[?(@.name=="VACUUM_ENABLED")].value}')
+    repack_enabled=$(kubectl get -n "$ns" cronjobs.batch --ignore-not-found maintenancejob -o jsonpath='{.spec.jobTemplate.spec.template.spec.containers[0].env[?(@.name=="REPACK_ENABLED")].value}')
+    if [ "$vacuum_enabled" == "true" ] && [ "$repack_enabled" == "false" ]; then
+      break
+    fi
+    echo "Waiting for envs True False"
```
Comment (Kidswiss) [inline tests/e2e/common/scripts/trigger-maintenance.sh:31]:
```suggestion
    echo "Waiting for envs vacuum: True repack: False"
```

File: tests/e2e/common/scripts/trigger-maintenance.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    sleep 15
+  done
+elif [ "$to_be_found" == "r" ]; then
+  while true; do
+    vacuum_enabled=$(kubectl get -n "$ns" cronjobs.batch --ignore-not-found maintenancejob -o jsonpath='{.spec.jobTemplate.spec.template.spec.containers[0].env[?(@.name=="VACUUM_ENABLED")].value}')
+    repack_enabled=$(kubectl get -n "$ns" cronjobs.batch --ignore-not-found maintenancejob -o jsonpath='{.spec.jobTemplate.spec.template.spec.containers[0].env[?(@.name=="REPACK_ENABLED")].value}')
+    if [ "$vacuum_enabled" == "false" ] && [ "$repack_enabled" == "true" ]; then
+      break
+    fi
+    echo "Waiting for envs False True"
```
Comment (Kidswiss) [inline tests/e2e/common/scripts/trigger-maintenance.sh:41]:
```suggestion
    echo "Waiting for envs vacuum: False repack: True"
```

File: tests/e2e/postgresql/02-connect.yaml [hunk 7]
```diff
@@ -3,7 +3,7 @@ kind: Job
 metadata:
   name: connect-postgresql
 spec:
-  backoffLimit: 1
+  backoffLimit: 6
```
Comment (Kidswiss) [inline tests/e2e/postgresql/02-connect.yaml:6]: Did you adjust this by mistake? The pod itself should not restart anymore, as the check script will simply loop.

File: tests/e2e/postgresql/scripts/trigger-maintenance.sh [hunk 8]
```diff
@@ -0,0 +1 @@
+../../../e2e/common/scripts/trigger-maintenance.sh
```
Comment (Kidswiss) [inline tests/e2e/postgresql/scripts/trigger-maintenance.sh:1]: I wouldn't put the script in the common folder. This script is very PostgreSQL specific.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
A few things to look at.

File: tests/e2e/postgresql/02-connect.yaml (hunk 7 above)
Comment (wejdross) [reply tests/e2e/postgresql/02-connect.yaml:6]: yes, that's leftover from past solution

**Comment by wejdross** [comment]
@Kidswiss all suggestions addressed and resolved

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**